tests/publiccloud/storage_perf.pm

```diff
@@ -0,0 +1,140 @@
+# SUSE's openQA tests
+#
+# Copyright © 2018 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Use FIO tool to run storage performance test
+#
+# Maintainer: Jose Lausuch <jalausuch@suse.com>
+
+use base "publiccloud::basetest";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use repo_tools 'generate_version';
+
+use constant NUMJOBS => 4;
+use constant IODEPTH => 4;
+
+sub run {
+    my ($self) = @_;
+    my $reg_code     = get_required_var('SCC_REGCODE');
+    my $runtime      = get_var('RUNTIME', 300);
+    my $block_device = get_var('BLOCK_DEVICE', '/dev/sdb');
+
+    my @scenario = (
+        {
+            name      => 'reference',
+            rw        => 'randread',
+            rwmixread => '100',
+            bs        => '8k'
+        },
+        {
+            name      => 'reallife',
+            rw        => 'randrw',
+            rwmixread => '65',
+            bs        => '8k'
+        },
+        {
+            name      => 'writeintensive',
+            rw        => 'randwrite',
+            rwmixread => '10',
+            bs        => '8k'
+
+        },
+        {
+            name      => 'maxthroughput',
+            rw        => 'read',
+            rwmixread => '100',
+            bs        => '64k'
+        }
+    );
+    $self->select_serial_terminal;
+
+    my $provider = $self->provider_factory();
+    my $instance = $provider->create_instance();
+    $instance->run_ssh_command(cmd => 'sudo SUSEConnect -r ' . $reg_code, timeout => 600);
```

Comment:
isn't this limit us to use only unregistered images ?

Reply:
I guess we will run this tests only for BYOS images...

Reply:
why?

Reply:
BYOS are the first images we get. The goal is to run performance during long time (not only 300s) and if we do this for all the possible images it would take too much time, and I don't expect a lot of performance differences between BYOS and ON_DEMAND images. But I might be wrong.

Reply:
for me it is not about should we or not run this on something different then BYOS . It is just about that we CAN do it . So skipping discussion around can you please make SCC_REGCODE not required in line 24 and run this line only in case SCC_REGCODE is defined ?

Reply:
ok